pkg/services/sqlstore/migrations/dashboard_mig.go

```diff
@@ -150,4 +150,18 @@ func addDashboardMigration(mg *Migrator) {
 	mg.AddMigration("Add column has_acl in dashboard", NewAddColumnMigration(dashboardV2, &Column{
 		Name: "has_acl", Type: DB_Bool, Nullable: false, Default: "0",
 	}))
+
+	// new uid column
+	mg.AddMigration("Add column uid in dashboard", NewAddColumnMigration(dashboardV2, &Column{
+		Name: "uid", Type: DB_NVarchar, Length: 12, Nullable: true,
+	}))
+
+	mg.AddMigration("Set uid column values", new(RawSqlMigration).
+		Sqlite("UPDATE dashboard SET uid=printf('%09d',id) WHERE uid IS NULL;").
+		Postgres("UPDATE dashboard SET uid=lpad('' || id,9,'0') WHERE uid IS NULL;").
+		Mysql("UPDATE dashboard SET uid=lpad(id,9,'0') WHERE uid IS NULL;"))
+
+	mg.AddMigration("Add index for uid in dashboard", NewAddIndexMigration(dashboardV2, &Index{
+		Cols: []string{"uid"}, Type: UniqueIndex,
+	}))
```

Comment:
One question here is whether we want to allow people to import the same dashboard into multiple orgs without having to modify the uid, in which case it might make more sense to have this index be on `org_id, uid`.  Not sure whether that's desirable or not.

Reply:
That's a valid point. From one perspective it would be really nice if dashboard url's are truly unique for a Grafana instance which means that you can jump into a dashboard in another organization without needing to provide OrgId as querystring. 

Unsure of how many users that will hit this use case taking into account new folder feature and if many users will migrate from organizations to folders? We may be fine with handling this in the import step. It will however generate a new uid then. 

We'll try to discuss this in the morning and comment here to loop you in. Feel free to comment on my comment in the mean time.

Reply:
@DanCech we decided to change the unique index to include org_id